test/e2e/e2e_suite_test.go

```diff
@@ -80,8 +85,33 @@ var _ = AfterSuite(func() {
 	Expect(err).NotTo(HaveOccurred())
 
 	if !clusterIsClean(clusterStateBefore, clusterStateAfter) {
-		log.Info("cluster resources before", "clusterStateBefore", clusterStateBefore)
-		log.Info("cluster resources after", "clusterStateAfter", clusterStateAfter)
+		pretty.Print(pretty.Diff(clusterStateBefore, clusterStateAfter))
+
+		var prettyJSON bytes.Buffer
+		var bmap, amap []byte
+		var err error
+		bmap, err = json.Marshal(map[string]ClusterResourceList{"ClusterStateBefore": clusterStateBefore})
+		if err != nil {
+			log.Error(err, "Problem with marshal")
+		} else {
+			err = json.Indent(&prettyJSON, bmap, "", "    ")
+			if err == nil {
+				log.Info("cluster resources before")
+				pretty.Print(string(prettyJSON.Bytes()))
+			}
+		}
```

Comment:
```suggestion
		if err != nil {
			log.Error(err, "Problem with marshal")
			
			return
		}

		err = json.Indent(&prettyJSON, bmap, "", "    ")
		if err == nil {
			log.Info("cluster resources before")
			pretty.Print(string(prettyJSON.Bytes()))
		}
```

Similar changes are recommended below.